src/engine/config.rs

```diff
@@ -1129,6 +1129,36 @@ impl Type {
         }
     }
 
+    /// Creates a new Type struct from a yaml-formatted string.
+    ///
+    /// # Arguments
+    ///
+    /// * yaml - yaml-formatted string containing Type definition
+    ///
+    /// # Errors
+    ///
+    /// Returns an [`Error`] variant [`DeserializationFailed`] if the yaml-formatted
+    /// string is improperly formatted.
+    ///
+    /// [`DeserializationFailed`]: ../../error/enum.Error.html#variant.DeserializationFailed
+    /// [`Error`]: ../../error/enum.Error.html
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// # use warpgrapher::engine::config::{Type};
+    ///
+    /// let t = Type::from_yaml("
+    /// name: User
+    /// props:
+    ///   - name: name
+    ///     type: String
+    /// ").unwrap();
+    /// ```
+    pub fn from_yaml(yaml: &str) -> Result<Type, Error> {
```

Comment:
Given that this is the only way to get a type from a &str, would it make sense to make this a TryFrom<&str> implementation instead?

Reply:
I would say so.

Reply:
Done.

Reply:
Done.